controller/src/lib.rs

```diff
@@ -20,114 +21,184 @@ pub mod plan;
 pub mod run;
 mod scheduler;
 
+type LogicalTime = u64;
+async fn spawn_backend(
+    nats: &TypedNats,
+    drone: &DroneId,
+    schedule_request: &ScheduleRequest,
+) -> anyhow::Result<(ScheduleResponse, Option<LogicalTime>)> {
```

Comment:
It looks like the `Option<LogicalTime>` is never actually used, am I missing something?

Reply:
Yea, it's used later for get_listener().

Reply:
Which PR is that in?

Reply:
This PR looks good otherwise but I'd like to see how the `Option<LogicalTime>` part is used in context because I generally think returning tuples is a code smell and suspect there may be a better way.

I suggest this path forward:
- Change this function to return a plain `anyhow::Result<ScheduleResponse>`. If you think this will take more than 5 minutes let me know.
- Request a review again and I'll LGTM it.
- In another PR, add the message type to request an existing lock without spawning (I'd imagine that PR will benefit from this refactoring)
- Once that's done we can have a design discussion about locking. It might be worth going straight to doing a two-phase commit.